### PR TITLE
Localization cleanup

### PR DESCRIPTION
**Pull request by melytc**
The objective of this PR is to:
- stop localizing rules that never show up in the UI.
- ensure that the rules that do show up, like `BrowseObjects `and `ProjectItemSchemas`, have the corresponding attributes filled.

This will help us to achieve two different things: first, avoid generating unnecessary work to the localization team; and second, ensure rule files have a standard on what attributes they need have to be properly localized.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/Items/None.BrowseObject.xaml [hunk 1]
```diff
@@ -1,16 +1,13 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!--Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.-->
+<!--NO_TRANSLATE-->
```
Comment (davidwengier) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/Items/None.BrowseObject.xaml]: Any item rules with BrowseObject in the name show up in the properties window for items of that type, so presumably do need translating.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralConfiguredBrowseObject.xaml [hunk 2]
```diff
@@ -1,16 +1,13 @@
...
-      DisplayName="General"
       OverrideMode="Replace"
       PageTemplate="generic"
       PropertyPagesHidden="true"
       xmlns="http://schemas.microsoft.com/build/2009/properties">
   <Rule.Categories>
-    <Category Name="General"
-              Description="General"
-              DisplayName="General" />
+    <Category Name="General" />
```
Comment (davidwengier) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralConfiguredBrowseObject.xaml]: If a rule doesn't show in the UI then it doesn't need categories at all, they are purely a UI concept.

File: tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/MiscellaneousRuleTests.cs [hunk 3]
```diff
@@ -30,13 +31,90 @@ public void NonVisiblePropertiesShouldntBeLocalized(string ruleName, string full
...
+                Assert.Null(element.Attribute("Description"));
+                Assert.Null(element.Attribute("Category"));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetAllDisplayedRules))]
+        public void BrowseObjectsAndProjectItemsSchemasShouldBeLocalized(string ruleName, string fullPath)
+        {
+            if(ruleName.Contains("ProjectItemsSchema") || (ruleName.Contains("BrowseObject") && !(ruleName.Contains("GeneralConfiguredBrowseObject")) && !(ruleName.Contains("None.BrowseObject")) ))
```
Comment (davidwengier) [inline tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/Rules/MiscellaneousRuleTests.cs]: None.BrowseObject is a real rule for items of type "None"

**Comment by melytc** [comment]
@davidwengier I have come up with another approach so I will close this PR and open a new one, I'll incorporate your feedback there :) thanks!

**Pull request closed without merging**